source/core/parameters.cc

```diff
@@ -291,6 +291,62 @@ namespace Parameters
     prm.leave_subsection();
   }
 
+  void
+  InterfaceSharpening::declare_parameters(ParameterHandler &prm)
+  {
+    prm.enter_subsection("interface sharpening");
+    {
+      prm.declare_entry("sharpening threshold",
+                        "0.5",
+                        Patterns::Double(),
+                        "VOF interface sharpening threshold");
+      // This parameter must be larger than 1 for interface sharpening. Choosing
+      // values less than 1 leads to interface smoothing instead of sharpening.
+      prm.declare_entry("interface sharpness",
+                        "2",
+                        Patterns::Double(),
+                        "VOF interface sharpness");
+      prm.declare_entry("sharpening frequency",
+                        "1000",
+                        Patterns::Integer(),
+                        "VOF interface sharpening frequency");
+      prm.declare_entry(
+        "verbosity",
+        "quiet",
+        Patterns::Selection("quiet|verbose"),
+        "State whether from the interface sharpening calculations should be printed "
+        "Choices are <quiet|verbose>.");
+
+      prm.declare_entry(
+        "l2 smoothing factor",
+        "0.000001",
+        Patterns::Double(),
+        "The smoothing factor for phase fraction L2 projection");
+    }
+    prm.leave_subsection();
+  }
+
+  void
+  InterfaceSharpening::parse_parameters(ParameterHandler &prm)
+  {
+    prm.enter_subsection("interface sharpening");
+    {
+      sharpening_threshold = prm.get_double("sharpening threshold");
+      interface_sharpness  = prm.get_double("interface sharpness");
+      sharpening_frequency = prm.get_integer("sharpening frequency");
```

Comment:
Should any sanity checking be done at this point?

Reply:
There is a warning for interface sharpness (<1 leads to interface smoothing instead of sharpening).
I've added two assertions for sharpening threshold and frequency.